launch/client.py

```diff
@@ -732,29 +735,27 @@ def edit_model_endpoint(
         )  # Returned from server as "creation"
         logger.info("Endpoint edit task id is %s", endpoint_creation_task_id)
 
-    def model_endpoint_exists(self, endpoint_name: str) -> bool:
-        for existing_endpoint in self.list_model_endpoints():
-            if endpoint_name == existing_endpoint.model_endpoint.name:
-                return True
-        return False
-
     def get_model_endpoint(
-        self, endpoint_name: str
+        self,
+        endpoint_name: str,
+        quiet: bool = False,
     ) -> Optional[Union[AsyncEndpoint, SyncEndpoint]]:
         """
         Gets a model endpoint associated with a name.
 
         Parameters:
             endpoint_name: The name of the endpoint to retrieve.
+            quiet: If ``True``, will return ``None`` without logging an error when the endpoint does not exist.
         """
         try:
             resp = self.connection.get(
                 os.path.join(ENDPOINT_PATH, endpoint_name)
             )
         except APIError:
-            logger.exception(
-                "Got an error when retrieving endpoint %s", endpoint_name
-            )
+            if not quiet:
```

Comment:
Wonder if we can use the status code to decide whether to log the error, i.e. if an endpoint doesn't exist, the HTTP status code returned should be a 404 (if the Gateway is implemented correctly anyways), and in that case `get_model_endpoint` should probably return None.

Reply:
2 questions: 1) do we always want to hide the 404 error log from users? - actually I think sometimes it's better to let user know? 2) for all the other errors/status code, do we want to return None as well or just throw an error? - I feel like it's better to not hide errors in other cases, tho I'm not sure what other errors can be

Reply:
I feel like if we return None from the method that's a good way to let the user know that the endpoint doesn't exist (which is what the 404 should mean), the other error codes we can just throw on (and not hide them) for now at least. Wonder what other opinions people have

Reply:
Yeah I don't really like or think we need this `quiet` flag, and I agree that the absence of a returned value is a suitable indicator that the endpoint doesn't exist (hence the `Optional` return type). This is because the endpoint not existing is an "expected" error case.

Reply:
the current scaleapi gateway is returning 500 instead of 404 in this case, it's better to fix that first and use the correct status code to decide logging or not